test/unit/deadline_client/cli/test_cli_bundle.py

```diff
@@ -290,6 +291,52 @@ def test_cli_bundle_job_name(fresh_deadline_config):
     assert result.exit_code == 0
 
 
+def test_cli_bundle_storage_profile_id(fresh_deadline_config):
+    """
+    Confirm that --storage-profile-id sets the ID that the job is submitted with, but does not
+    change the value of storage profile saved to the configuration file.
+    """
+    PRE_STORAGE_PROFILE_ID = "sp-11223344556677889900abbccddeeff"
+    CLI_STORAGE_PROFILE_ID = "sp-0000000000000000000000000000000"
+
+    config.set_setting("defaults.farm_id", MOCK_FARM_ID)
+    config.set_setting("defaults.queue_id", MOCK_QUEUE_ID)
+
+    # Set the storage profile ID in the config; as someone may have by using `deadline config set`
+    config.set_setting("settings.storage_profile_id", PRE_STORAGE_PROFILE_ID)
+
+    # Use a temporary directory for the job bundle
+    with tempfile.TemporaryDirectory() as tmpdir, patch.object(boto3, "Session") as session_mock:
+        session_mock().client("deadline").create_job.return_value = MOCK_CREATE_JOB_RESPONSE
+        session_mock().client("deadline").get_job.return_value = MOCK_GET_JOB_RESPONSE
+        session_mock.reset_mock()
+
+        # Write a JSON template
+        with open(os.path.join(tmpdir, "template.json"), "w", encoding="utf8") as f:
+            f.write(MOCK_JOB_TEMPLATE_CASES["MINIMAL_JSON"][1])
+
+        runner = CliRunner()
+        with patch.object(api_module, "get_storage_profile_for_queue"):
+            result = runner.invoke(
+                main,
+                ["bundle", "submit", tmpdir, "--storage-profile-id", CLI_STORAGE_PROFILE_ID],
+            )
+
+    assert tmpdir in result.output
+    assert MOCK_CREATE_JOB_RESPONSE["jobId"] in result.output
+    assert MOCK_GET_JOB_RESPONSE["lifecycleStatusMessage"] in result.output
+    session_mock().client().create_job.assert_called_once_with(
+        farmId=MOCK_FARM_ID,
+        queueId=MOCK_QUEUE_ID,
+        template=ANY,
+        templateType="JSON",
+        priority=50,
+        storageProfileId=CLI_STORAGE_PROFILE_ID,
+    )
+    assert result.exit_code == 0
+    assert get_setting("settings.storage_profile_id") == CLI_STORAGE_PROFILE_ID
```

Comment:
Mmm... Should this not be `PRE_STORAGE_PROFILE_ID` if it wasn't changed due to the command line arg?

Reply:
Right you are.